### PR TITLE
MAINT: Add Azure

### PR DESCRIPTION
**Pull request by larsoner**
see #262

### CODE REVIEW

**Comment by jasmainak** [comment]
Just gave you access to autoreject

**Comment by larsoner** [comment]
Argh, I actually need organization-level access to `github.com/autoreject/` in order to install https://github.com/marketplace/azure-pipelines

**Comment by jasmainak** [comment]
just sent !

**Comment by larsoner** [comment]
@hoechenberger everything is configured at the Azure end. In case you want to see the email stuff:

https://dev.azure.com/autoreject/autoreject/_settings/notifications#

All I did was add you two / your emails to the `autoreject` team, click the `+ New subscription` at the top, and set the conditionals that you see in `A scheduled build fails`. 

Now we just wait for them to approve our OSS pipeline credits, which I requested [here](https://aka.ms/azpipelines-parallelism-request) (as their error message [suggests](https://dev.azure.com/autoreject/autoreject/_build/results?buildId=2&view=logs&j=c2fabf20-0dae-5771-2529-62cd5b113ff9)). I'm guessing that it didn't come with some to start (?) because when I went to link GitHub to Azure, I didn't have the right project permissions at the Azure end. (I've never had this problem when setting up both from my account.)

**Comment by larsoner** [comment]
Okay this is working. They gave us only 1 parallel job, and it has to be private, which means only me @hoechenberger and @jasmainak can actually look at the build logs. I sent them an email about trying to get the Azure builds to be public, we'll see what they say. In a worst case, if they don't respond, builds start failing and people can't see why, we can disable it. But for now, the stable and dev builds only take ~5 min each, and the scheduled build should be all set up now, so I think it's worth trying.

Ready for review/merge from my end @jasmainak !

**Comment by jasmainak** [comment]
do I need to create an account on Azure to see the logs?

also just for my clarification, Azure was the choice for testing against MNE nightly because that's the pipeline you already had set up? I guess this could be done with github actions too if Azure became problematic?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,19 +1,60 @@
...
 trigger:
-- master
-
-pool:
-  vmImage: ubuntu-latest
+  # start a new build for every push
+  batch: False
+  branches:
+    include:
+      - 'main'
```
Comment (sappelhoff) [inline azure-pipelines.yml:6]: it's still called "master" on autoreject :-)

**Comment by larsoner** [comment]
> do I need to create an account on Azure to see the logs?

Yes, for now at least

>  Azure was the choice for testing against MNE nightly because that's the pipeline you already had set up?

It's because it's the one where I knew how to email individual people when the scheduled build fails. Most CI providers have ways of doing scheduled builds, but I'm not sure which others allow you to email specific people when the scheduled build fails. @hoechenberger looked into it for GH actions a little bit at least, and I'm it sounds like he didn't find a way to do it.

File: azure-pipelines.yml (hunk 1 above)
Comment (sappelhoff) [reply azure-pipelines.yml]: fine from my side - I am done with all changes I wanted to do and won't interfere :-)

**Comment by jasmainak** [comment]
Okay understood. Could someone point me to where I should sign up? Do I need an invite from the autoreject org on Azure? I don't think I got it or it went to my spam or something ...

**Comment by larsoner** [comment]
Hopefully you now have an email at your gmail.com address

**Comment by larsoner** [comment]
... if you can see a "A scheduled build fails" on this page you'll know it's working:

https://dev.azure.com/autoreject/autoreject/_settings/notifications

**Comment by jasmainak** [comment]
nopes, no email. Did you send to mainakjas [at] gmail [dot] com ?

**Comment by larsoner** [comment]
Yes indeed. Perhaps just try making an Azure account using that email, then see if you can access that URL above? Maybe you have to have an account before I try to add you...

**Comment by larsoner** [comment]
I assume this URL is one you could use to sign up https://azure.microsoft.com/en-us/free/ . Or try to navigate to https://dev.azure.com/autoreject/autoreject and it should ask you to sign in or create an account -- do so with your gmail email address and hopefully it lets you in!

**Comment by jasmainak** [comment]
Do you mind sending the invite to mjas [at] mgh [dot] harvard [dot] edu? I signed up to Azure with it. I don't have a microsoft account associated with gmail, not sure if I want to create a new email ID for microsoft ...

**Comment by larsoner** [comment]
Done!

**Comment by jasmainak** [comment]
Thanks @larsoner !

**Comment by larsoner** [comment]
FYI they said they should be able to convert us to public builds tomorrow. I'll adjust settings once they do!

**Pull request merged**